components/inputs/input-text.js

```diff
@@ -367,8 +378,7 @@ class InputText extends FormElementMixin(SkeletonMixin(RtlMixin(LitElement))) {
 						style="${styleMap(inputStyles)}"
 						tabindex="${ifDefined(this.tabindex)}"
 						title="${ifDefined(this.title)}"
-						type="${this._getType()}"
-						.value="${this.value}">
```

Comment:
I've removed the binding to the input's `value` altogether. Otherwise anytime a re-render is triggered by anything -- validation state change, value change, mouse hover, etc. -- the cursor will jump to the end in Safari.

The workaround is instead to manually reach into the DOM and set the input's value when we initially render (in `firstUpdated`) and whenever the property setter is set (including via attribute).